addons/slack/template/index.js

```diff
@@ -0,0 +1,58 @@
+const slack = require('./slack.js');
+const Sdk = require('@fusebit/add-on-sdk');
+const { WebClient } = require('@slack/web-api');
+const Superagent = require('superagent');
+
+module.exports = async (ctx) => {
+  const sdk = {
+    bot: new WebClient(ctx.configuration.slack_bot_access_token),
+    user: ctx.configuration.slack_user_access_token ? new WebClient(ctx.configuration.slack_user_access_token) : null,
+  };
+
+  // TODO What prevents someone who finds out this URL to
+  // send unauthorized events?
+
+  if (ctx.url.endsWith('events/ingest')) {
+    Sdk.debug('Received Events API event');
+
+    let response;
+    try {
+      if (ctx.method === 'POST') {
+        let event = ctx.body;
+        Sdk.debug('Valid event', event);
+
+        // Attach all event handlers off a nested `slack` property
+        let handlers = slack.slack;
+        if (handlers) {
+          let functions = Object.getOwnPropertyNames(handlers).filter((p) => typeof handlers[p] === 'function');
+          let toInvoke = functions.filter((f) => f == event.type || f == 'all');
+
+          Sdk.debug('Invoking handlers', toInvoke);
+          toInvoke.forEach((t) => {
+            handlers[t](sdk, event, {
+              ...ctx.configuration,
+              storage: Sdk.getStorageClient(ctx),
+            });
+          });
+        }
+
+        return {
+          status: 201,
+        };
+      } else {
+        return {
+          status: 400,
+          body: { message: 'Unsupported HTTP request type' },
+        };
+      }
+    } catch (e) {
+      Sdk.debug('Call to registration endpoint failed', e);
+      return { status: 500 };
+    }
+  } else {
+    return slack(sdk, {
+      ...ctx.configuration,
+      storage: Sdk.getStorageClient(ctx),
+    });
```

Comment:
What is the purpose of this call? What is the invocation if this is not "events/ingest"?